File: benchmark/addon.go
package main

import (
	"github.com/deflix-tv/go-stremio"
)

var (
	manifest = stremio.Manifest{
		ID:      "org.myexampleaddon",
		Version: "1.0.0",

		Description: "simple example",
		Name:        "simple example",

		Catalogs: []stremio.CatalogItem{},
		ResourceItems: []stremio.ResourceItem{
			{
				Name:  "stream",
				Types: []string{"movie"},
			},
		},
		Types:      []string{"movie"},
		IDprefixes: []string{"tt"},
	}
)

func streamHandler(id string) ([]stremio.StreamItem, error) {
	if id == "tt1254207" {
		return []stremio.StreamItem{{URL: "http://distribution.bbb3d.renderfarming.net/video/mp4/bbb_sunflower_1080p_30fps_normal.mp4"}}, nil
	}
	return nil, stremio.NotFound
}

func main() {
	streamHandlers := map[string]stremio.StreamHandler{"movie": streamHandler}

	addon, err := stremio.NewAddon(manifest, nil, streamHandlers, stremio.Options{BindAddr: "0.0.0.0", Port: 7000, DisableRequestLogging: true})
	if err != nil {
		addon.Logger().Sugar().Fatalf("Couldn't create addon: %v", err)
	}

	addon.Run()
}

File: benchmark/addon.js
#!/usr/bin/env node

const { addonBuilder, serveHTTP, publishToCentral } = require('stremio-addon-sdk')

const builder = new addonBuilder({
    id: 'org.myexampleaddon',
    version: '1.0.0',

    name: 'simple example',

    catalogs: [],
    resources: ['stream'],
    types: ['movie'],
    idPrefixes: ['tt']
})

builder.defineStreamHandler(function(args) {
    if (args.type === 'movie' && args.id === 'tt1254207') {
        const stream = { url: 'http://distribution.bbb3d.renderfarming.net/video/mp4/bbb_sunflower_1080p_30fps_normal.mp4' }
        return Promise.resolve({ streams: [stream] })
    } else {
	return Promise.resolve({ streams: [] })
    }
})

serveHTTP(builder.getInterface(), { port: 7000 })
